Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Driver.hs

```diff
@@ -1088,33 +1101,51 @@ getFleetDriverAssociation ::
+getFleetDriverAssociation merchantShortId _opCity fleetOwnerId mbLimit mbOffset mbCountryCode mbDriverPhNo mbIsActive mbStats mbFrom mbTo mbMode = do
   merchant <- findMerchantByShortId merchantShortId
-  listOfAllDrivers <- getListOfDrivers mbCountryCode mbDriverPhNo fleetOwnerId merchant.id mbIsActive mbLimit mbOffset
+  listOfAllDrivers <- getListOfDrivers mbCountryCode mbDriverPhNo fleetOwnerId merchant.id mbIsActive mbLimit mbOffset mbMode
   listItems <- createFleetDriverAssociationListItem listOfAllDrivers
-  pure $ Common.DrivertoVehicleAssociationRes {fleetOwnerId = fleetOwnerId, listItem = listItems}
+  let summary = Common.Summary {totalCount = 10000, count = length listItems}
+  pure $ Common.DrivertoVehicleAssociationRes {fleetOwnerId = fleetOwnerId, listItem = listItems, summary = summary}
   where
     createFleetDriverAssociationListItem :: (CacheFlow m r, EsqDBFlow m r, EncFlow m r, CH.HasClickhouseEnv CH.APP_SERVICE_CLICKHOUSE m) => [FleetDriverAssociation] -> m [Common.DriveVehicleAssociationListItem]
     createFleetDriverAssociationListItem fdaList = do
+      now <- getCurrentTime
+      let defaultFrom = UTCTime (utctDay now) 0
+          from = fromMaybe defaultFrom mbFrom
+          to = fromMaybe now mbTo
       forM fdaList $ \fda -> do
         driver <- QPerson.findById fda.driverId >>= fromMaybeM (PersonNotFound fda.driverId.getId)
-        driverActiveRcAssociation <- QRCAssociation.findActiveAssociationByDriver fda.driverId True
-        (vehicleNo, vehicleType) <- case driverActiveRcAssociation of
-          Just rcAssociation -> do
-            isAssociatedWithFleet <- RCQuery.findByRCIdAndFleetOwnerId rcAssociation.rcId (Just fleetOwnerId)
-            case isAssociatedWithFleet of
-              Just _ -> do
-                decryptedVehicleRC <- RCQuery.findById rcAssociation.rcId >>= fromMaybeM (RCNotFound rcAssociation.rcId.getId)
-                vehNo <- decrypt (decryptedVehicleRC.certificateNumber)
-                pure (Just vehNo, castVehicleVariantDashboard decryptedVehicleRC.vehicleVariant)
-              Nothing -> pure (Nothing, Nothing)
-          Nothing -> pure (Nothing, Nothing)
+        driverRCAssociation <- QRCAssociation.findAllByDriverId fda.driverId
+        let rcAssociatedWithFleet = filter (\(_, rc) -> rc.fleetOwnerId == Just fleetOwnerId) driverRCAssociation
+        (vehicleNo, vehicleType) <- case rcAssociatedWithFleet of ---- so the logic is if it have active association with the fleet vehicle return that otherwise return the latest one
+          [] -> pure (Nothing, Nothing)
+          associations -> do
+            let activeAssociation = find (\(assoc, _) -> assoc.isRcActive) associations
+            case activeAssociation of
+              Just (_, rc) -> getVehicleDetails rc ------- if driver is using fleet vehicle
+              Nothing -> getVehicleDetails $ snd $ head associations -------- otherwise give the latest active association
```

Comment:
can you avoid partial `head` function?

Reply:
since associations will never be empty as empty case is already handled above therefore used that.